### PR TITLE
#871 - show pop-up menu for taxon photo - share / view in browser

### PR DESCRIPTION
**Pull request by budowski**

### CODE REVIEW

File: src/components/TaxonDetails/TaxonDetails.js [hunk 1]
```diff
@@ -122,6 +141,33 @@ const TaxonDetails = ( ): Node => {
...
+              onPress={( ) => {
+                openURLInBrowser( taxonUrl );
+                setKebabMenuVisible( false );
+              }}
+              title={t( "View-in-browser" )}
+            />
+            <Menu.Item
+              onPress={( ) => {
+                Share.share( {
+                  message: taxonUrl,
```
Comment (albullington) [inline src/components/TaxonDetails/TaxonDetails.js:162]: On iOS, having both message and url keys results in text that says "2 Links", as shown below. We solved this on ObsDetails using a [conditional statement](https://github.com/inaturalist/iNaturalistReactNative/blob/94ea2b6054dc06a0fea8e234e28a525ff08377ec/src/components/ObsDetails/HeaderKebabMenu.js#L26) for iOS vs. Android.

Screenshot 2023-12-18 at 2 44 28 PM

File: src/components/TaxonDetails/TaxonDetails.js [hunk 2]
```diff
@@ -64,6 +68,7 @@ const TaxonDetails = ( ): Node => {
     logger.error( `Failed to retrieve taxon ${id}: ${error}` );
   }
   const taxon = remoteTaxon || localTaxon;
+  const taxonUrl = `${TAXON_URL}/${taxon.id}`;
```
Comment (albullington) [inline src/components/TaxonDetails/TaxonDetails.js:71]: This line is throwing an error for me on Android.

Screenshot 2023-12-18 at 2 46 18 PM

**Review by albullington — CHANGES_REQUESTED** [review]
Blockers:
1. The 2 links issue on iOS and
2. The null error that's preventing me from opening TaxonDetails at all on Android

A nice-to-have would be adding a few tests, similar to what we have for the [kebab/share menu on ObsDetails](https://github.com/inaturalist/iNaturalistReactNative/blob/94ea2b6054dc06a0fea8e234e28a525ff08377ec/tests/unit/components/ObsDetails/HeaderKebabMenu.test.js#L25), to make sure the kebab menu opens the expected browser link and the expected share link on iOS/Android.

**Comment by budowski** [comment]
Thanks - but very weird with Android, since I had it working for me - what steps do you take to recreate this? How do you reach the taxon page and what taxon are you viewing?

**Comment by albullington** [comment]
> Thanks - but very weird with Android, since I had it working for me - what steps do you take to recreate this? How do you reach the taxon page and what taxon are you viewing?

I tried as a logged out user. Went to Explore -> ObsDetails for Yellow Admiral -> TaxonDetails which resulted in a crash.

**Comment by albullington** [comment]
fwiw, I believe all you need is the optional chaining operator here (i.e. `taxon?.id` instead of taxon.id) to prevent a crash -- JavaScript is complaining about looking for the id key on a null taxon object.

**Comment by budowski** [comment]
Thanks, but the problem is that I need to show some URL there, otherwise it will be a taxon URL without the actual taxon ID - I'll look into why that object is empty

**Pull request merged**